Figgle.Generator/RenderTextSourceGenerator.cs

```diff
@@ -97,244 +100,263 @@ public GenerateFiggleTextAttribute(string memberName, string fontName, string so
-
-                            if (namedTypeSymbol.ContainingType is not null)
-                            {
-                                Diagnostics.Add(Diagnostic.Create(
-                                    NestedTypeIsNotSupportedDiagnostic,
-                                    Location.Create(attribute.SyntaxTree, attribute.Span),
-                                    key.ClassName));
-                                continue;
-                            }
-
-                            if (!classDeclaration.Modifiers.Any(modifier => modifier.Text == "partial"))
-                            {
-                                Diagnostics.Add(Diagnostic.Create(
-                                    TypeIsNotPartialDiagnostic,
-                                    Location.Create(attribute.SyntaxTree, attribute.Span),
-                                    key.ClassName));
-                                continue;
-                            }
-
-                            if (!DataByType.TryGetValue(key, out var typeItems))
-                            {
-                                DataByType[key] = typeItems = new(new(), new(StringComparer.Ordinal));
-                            }
-
-                            typeItems.SeenMemberNames.UnionWith(namedTypeSymbol.MemberNames);
-
-                            var memberName = arg0.Token.ValueText;
-                            var memberNameLocation = ToLocation(attribute.ArgumentList.Arguments[0]);
-
-                            var fontName = arg1.Token.ValueText;
-                            var fontNameLocation = ToLocation(attribute.ArgumentList.Arguments[1]);
-
-                            var sourceText = arg2.Token.ValueText;
-
-                            if (!SyntaxFacts.IsValidIdentifier(memberName))
-                            {
-                                Diagnostics.Add(Diagnostic.Create(
-                                    InvalidMemberNameDiagnostic,
-                                    memberNameLocation,
-                                    memberName));
-                                continue;
-                            }
-
-                            typeItems.Items.Add(new(memberName, memberNameLocation, fontName, fontNameLocation, sourceText));
-
-                            static Location ToLocation(SyntaxNode node) => Location.Create(node.SyntaxTree, node.Span);
-                        }
-                    }
-                }
+                return true;
+            }
+
+            if (x is null || y is null)
+            {
+                return false;
             }
+
+            return x.MemberName == y.MemberName
+                && x.FontName == y.FontName
+                && x.SourceText == y.SourceText;
+        }
+
+        public int GetHashCode(RenderItem obj)
+        {
+            return (obj.MemberName, obj.FontName, obj.SourceText).GetHashCode();
```

Comment:
That's an elegant way to combine some hash codes without access to the `HashCode` class!